File: 第五次课/第5次课作业代码/第5次课作业代码/第四题/bvp_legendre/legendregauss.m
function z=legendregauss(n,m)

error=1e-14;
h=n^(-2);
a=-1;
b=a+h;

for k=1:n-m
    
    legendre_a=legendre(n,m,a);
    legendre_b=legendre(n,m,b);
    while(legendre_a*legendre_b>0)
        a=b;
        legendre_a=legendre_b;
        
        b=a+h;
        legendre_b=legendre(n,m,b);
    end
   
   x=(a+b)/2;
   xright=b;
   while(abs(x-xright)>error)
        xright=x;
        x=x-legendre(n,m,x)/legendre(n,m+1,x);
   end
   z(k)=x;
   a=x+h;
   b=a+h;
end


end

File: 第五次课/第5次课作业代码/第5次课作业代码/第一题/勒让德插值/python/legendregauss.py
import numpy as np
from lengendre import legendre
def legendregauss(n,m):
    if n==0:
        return []

    z=[]
    error=1e-14
    h=n**(-2)
    a=-1
    b=a+h
    number_newton=0;
    for k in range(1,n-m+1):
    
        legendre_a=legendre(n,m,a)
        legendre_b=legendre(n,m,b)
        while(legendre_a*legendre_b>0):
            a=b
            legendre_a=legendre_b
            
            b=a+h
            legendre_b=legendre(n,m,b)
    
        x=(a+b)*0.5
        xright=b
        while(np.abs(x-xright)>error):
            xright=x
            x=x-legendre(n,m,x)/legendre(n,m+1,x)
            number_newton+=1;
        z.append(x)
        a=x+h
        b=a+h
    return np.array(z)
# def legendregauss(n,m):
#     if n==0:
#         return []

#     z=[]
#     error=1e-14
#     h=n**(-2)
#     a=-1
#     b=a+h
    
#     for k in range(1,n-m+1):
    
#         legendre_a=legendre(n,m,a)
#         legendre_b=legendre(n,m,b)
#         while(legendre_a*legendre_b>0):
#             a=b
#             legendre_a=legendre_b
            
#             b=a+h
#             legendre_b=legendre(n,m,b)
    
#         x=(a+b)*0.5
#         xright=b
#         while(np.abs(x-xright)>error):
#             xright=x
#             x=x-legendre(n,m,x)/legendre(n,m+1,x)
    
#         z.append(x)
#         a=x+h
#         b=a+h
    
#     return np.array(z)